modules/connectivity/ParticipantConnectionStatus.js

```diff
@@ -0,0 +1,306 @@
+
+    this.rtc.addListener(
+        RTCEvents.ENDPOINT_CONN_STATUS_CHANGED,
+        this._onEndpointConnStatusChanged);
+
+    // On some browsers MediaStreamTrack trigger "onmute"/"onunmute"
+    // events for video type tracks when they stop receiving data which is
+    // often a sign that remote user is having connectivity issues
+    if (RTCBrowserType.isVideoMuteOnConnInterruptedSupported()) {
+
+        this._onTrackRtcMuted = this.onTrackRtcMuted.bind(this);
+        this.rtc.addListener(
+            RTCEvents.REMOTE_TRACK_MUTE, this._onTrackRtcMuted);
+
+        this._onTrackRtcUnmuted = this.onTrackRtcUnmuted.bind(this);
+        this.rtc.addListener(
+            RTCEvents.REMOTE_TRACK_UNMUTE, this._onTrackRtcUnmuted);
+
+        // Track added/removed listeners are used to bind "mute"/"unmute"
+        // event handlers
+        this._onRemoteTrackAdded = this.onRemoteTrackAdded.bind(this);
+        this.conference.on(
+            JitsiConferenceEvents.TRACK_ADDED, this._onRemoteTrackAdded);
+        this._onRemoteTrackRemoved = this.onRemoteTrackRemoved.bind(this);
+        this.conference.on(
+            JitsiConferenceEvents.TRACK_REMOVED, this._onRemoteTrackRemoved);
+
+        // Listened which will be bound to JitsiRemoteTrack to listen for
+        // signalling mute/unmute events.
+        this._onSignallingMuteChanged = this.onSignallingMuteChanged.bind(this);
+    }
+};
+
+/**
+ * Removes all event listeners and disposes of all resources held by this
+ * instance.
+ */
+ParticipantConnectionStatus.prototype.dispose = function () {
+
+    this.rtc.removeListener(
+        RTCEvents.ENDPOINT_CONN_STATUS_CHANGED,
+        this._onEndpointConnStatusChanged);
+
+    if (RTCBrowserType.isVideoMuteOnConnInterruptedSupported()) {
+        this.rtc.removeListener(
+            RTCEvents.REMOTE_TRACK_MUTE, this._onTrackRtcMuted);
+        this.rtc.removeListener(
+            RTCEvents.REMOTE_TRACK_UNMUTE, this._onTrackRtcUnmuted);
+        this.conference.off(
+            JitsiConferenceEvents.TRACK_ADDED, this._onRemoteTrackAdded);
+        this.conference.off(
+            JitsiConferenceEvents.TRACK_REMOVED, this._onRemoteTrackRemoved);
+    }
+
+    Object.keys(this.trackTimers).forEach(function (participantId) {
+        this.clearTimeout(participantId);
+    }.bind(this));
+};
+
+/**
+ * Checks whether given <tt>JitsiParticipant</tt> has any muted video
+ * <tt>MediaStreamTrack</tt>s.
+ *
+ * @param {JitsiParticipant} participant to be checked for muted video tracks
+ *
+ * @return {boolean} <tt>true</tt> if given <tt>participant</tt> contains any
+ * video <tt>MediaStreamTrack</tt>s muted according to their 'muted' field.
+ */
+var hasRtcMutedVideoTrack = function (participant) {
+    return participant.getTracks().some(function(jitsiTrack) {
+        var rtcTrack = jitsiTrack.getTrack();
+        return jitsiTrack.getType() === MediaType.VIDEO
+            && rtcTrack && rtcTrack.muted === true;
```

Comment:
I think this check should better fit as a method of JitsiTrack or JitsiRemoteTrack object instead of accessing rtcTrack.muted property which should be private.